### ISSUE TITLE
Compile BC against multiple versions 

### ISSUE THREAD

**Issue by borisimple** [HodorNV/ALOps#218]
Is there a way to use a specific app.json (if you have any) instead of standard app.json?
In the screenshot, you can see that the JSON object is always reading app.json file.

![image](https://user-images.githubusercontent.com/16119596/95455281-e7349400-096d-11eb-9a01-9f2d4690ae75.png)

The story here is that we temporarily need to compile and build project against v16.

We have one build pipeline which compiles the app against the latest available BC, which at the moment is 17.
I was thinking to create another build pipeline which will compile the app against V16, by using app-v16.json.
So in yaml, I reference correct app-v16.json file for compilation:

```
  - task: ALOpsAppCompiler@1
    displayName: Compiling the app $(app_name)
    inputs:
      usedocker: true
      targetproject: "app/app-v16.json"
      nav_app_version: "17.0.[yyyyww].*"
      app_file_suffix: _APP_V16
      failed_on_warnings: true
```

There is maybe a better way to do this, but I thought to give this a shot.

**Comment by waldo1001** [comment]
A better way to do this, is to just create 2 branches, both with their own version of the app.json, and set up to build in both pipelines.  You'll need to merge with the master, but this can be automated with an extra step in the pipeline of the sub-branch.

Makes sense?

**Comment by borisimple** [comment]
Yep, but I was thinking to skip merging those changes manually..
So it is time to figure out how to merge it automatically 😏

**Comment by waldo1001** [comment]
YOu don't have to do it manually - you can add another yaml-step for this.  Like this:

```yaml
- powershell: |
   $env:GIT_REDIRECT_STDERR = '2>&1' # Do not report git output as error... However error is also only text now
   
   git checkout master
   git pull origin master
   git checkout $env:branch_name
   git fetch
   git pull
   git pull origin master
   git push
  env:
    branch_name: '${{ parameters.branch_name }}'
  displayName: 'Update Branch From Master'
```

**Comment by waldo1001** [comment]
make sure you have git on the build server ;-)

**Comment by AdminHodor** [comment]
The import thing here is that ALC.EXE will compile based on a folder only, and will ALWAYS assume [app.json] is the core to start from, no reasonable way around it.  

So [targetproject: "app/app-v16.json"] will eventually always result in having the [app.json] from that folder processed.

I do have a suggestion that we could introduce without breaking other peoples pipelines:
- When we detect "targetproject" is not "app.json" , we'll brute-force copy it to "app.json" before compilation
- This way you could have a "app-v15.json" and a "app-v16.json", target them both, and @compilation we'll copy it to the expected file (thus also overwriting the original app,json that would be present)

Could this work for you? We're sure willing to give this a try.

Kind regards

**Comment by borisimple** [comment]
@AdminHodor 

I think this would be great. It would simplify the process a lot.

**Comment by AdminHodor** [comment]
Dear @borisimple ,

We just released v1.435.1856 which should tackle this issue.
Please note it could take a couple of pipeline runs for DevOps to pickup the new version.

Kind regards,

**Comment by borisimple** [comment]
@AdminHodor 

Work like a charm! 😉😏
